protocols/syslog/syslog.c

```diff
@@ -34,88 +35,142 @@
 #include "syslog_net.h"
 
 
-static char send_buffer[MAX_DYNAMIC_SYSLOG_BUFFER + 1];
+static char send_buffer[MAX_LINES_SYSLOG_BUFFER][MAX_LENGTH_SYSLOG_BUFFER + 1];
```

Comment:
I suggest to not hard-code the number of buffers and thus waste memory. Lets dynamically allocate and free it and define a maximum number of allocated buffers.